lib/java_buildpack/framework/fusion_reactor.rb

```diff
@@ -0,0 +1,123 @@
+        @component_name                                  = 'FusionReactor'
+        @application                                     = context[:application]
+        @configuration                                   = context[:configuration]
+        @droplet                                         = context[:droplet]
+        return unless supports?
+        @version = 'latest'
+        @uri, @lib_uri, @password, @with_debug = find_agent
+      end
+
+      # (see JavaBuildpack::Component::BaseComponent#compile)
+      def compile
+        download_jar self.jar_name, @droplet.sandbox, @component_name
+        return unless @with_debug
+
+        download(@version, @lib_uri, lib_name) do |file|
+          FileUtils.mkdir_p @droplet.sandbox
+          FileUtils.cp_r(file.path, @droplet.sandbox + lib_name)
+        end
+      end
+
+      # (see JavaBuildpack::Component::BaseComponent#release)
+      def release
+        service     = @application.services.find_service FILTER
+        credentials = service['credentials']
+        java_opts   = @droplet.java_opts
+
+        java_opts.add_system_property('frlicense', credentials[LICENSE_KEY])
+        java_opts.add_system_property('fradminpassword', @password)
+
+        credentials.each do |key, value|
+          next if SYSTEM_KEYS.include? key
+          java_opts.add_system_property(key, value)
+        end
+
+        props = {}
+        props['name'] = credentials['instance_name'] || @application.details['application_name']
+        props['address'] = credentials['instance_port'] || '8088'
+
+        java_opts.add_javaagent_with_props(@droplet.sandbox + jar_name, props)
+
+        return unless @with_debug
+        java_opts.add_agentpath(@droplet.sandbox + lib_name)
+      end
+
+      protected
+
+      # (see JavaBuildpack::Component::VersionedDependencyComponent#supports?)
+      def supports?
+        (architecture == 'x86_64' || architecture == 'i686') and (@application.services.one_service? FILTER, [LICENSE_KEY])
+      end
+
+      def jar_name
+        'fusionreactor.jar'
+      end
+
+      def lib_name
+        'libfrjvmti_x64.so'
+      end
+
+      private
+
+      FILTER = /fusionreactor/
+
+      LICENSE_KEY = 'license'
+
+      SYSTEM_KEYS = [LICENSE_KEY, 'version', 'password', 'debug', 'agent_download_url', 'lib_download_url']
+
+      private_constant :FILTER, :LICENSE_KEY
+
+      def architecture
+        `uname -m`.strip
+      end
+
+      def find_agent
+        service     = @application.services.find_service FILTER
+        credentials = service['credentials']
+
+        password   = credentials['password']
+        with_debug = credentials['debug']
+        uri        = credentials['agent_download_url']
+        lib_uri    = credentials['lib_download_url']
+
+        if with_debug.nil?
+          with_debug = true
+        end
+
+        unless uri
+            uri = 'https://intergral-dl.s3.amazonaws.com/FR/Latest/fusionreactor.jar'
```

Comment:
The reason we require one of our repositories is that sadly we can't support downloading from arbitrary locations at run time.  The vast majority of Cloud Foundry instances are run without access to the open internet and therefore cannot download agents from it.